src/NHibernate/Linq/Visitors/QueryModelVisitor.cs

```diff
@@ -27,6 +27,9 @@ public class QueryModelVisitor : NhQueryModelVisitorBase, INhQueryModelVisitor
 		public static ExpressionToHqlTranslationResults GenerateHqlQuery(QueryModel queryModel, VisitorParameters parameters, bool root,
 			NhLinqExpressionReturnType? rootReturnType)
 		{
+			// Expand conditionals in subquery FROM clauses into multiple subqueries
+			SubQueryConditionalExpander.ReWrite(queryModel);
+			
```

Comment:
Some undue trailing whitespaces are left on line 32. @hazzik, shouldn't we add in .editorconfig `trim_trailing_whitespace = true`? Of course it may at first generate some noise on edited files by removing all trailing whitespaces.